burger/toppings/pluginchannels.py

```diff
@@ -0,0 +1,163 @@
+
+        clientbound += all_channels[0]
+        serverbound += all_channels[1]
+
+def _require_fields(aggregate, required_fields):
+    for field in required_fields:
+        assert field in aggregate, f"{field} is missing from aggregate"
+
+        for subfield in required_fields[field]:
+            assert subfield in aggregate[field], f"{field}.{subfield} is mising from aggregate"
+
+def _get_version_info(aggregate):
+    protocol = aggregate["version"].get("protocol")
+    netty_rewrite = aggregate["version"].get("netty_rewrite")
+    distribution = aggregate["version"].get("distribution")
+    
+    return (protocol, netty_rewrite, distribution)
+
+def _get_custom_payload_packets(classloader, ignore_clientbound=False, ignore_serverbound=False):
+    clientbound_packet = None
+    serverbound_packet = None
+
+    for class_name in classloader.classes:
+
+        if (ignore_clientbound or clientbound_packet is not None) and (ignore_serverbound or serverbound_packet is not None):
+            break
+
+        constants = _get_class_constants(classloader, class_name)
+
+        # Make sure we have the right message, and at least one identifier declared in the class (to avoid login custom payload packet)
+        if not ignore_clientbound and clientbound_packet is None and "Payload may not be larger than 1048576 bytes" in constants:
+            if any([const for const in constants if _is_channel_identifier(const)]):
+                clientbound_packet = class_name
+        elif not ignore_serverbound and serverbound_packet is None and "Payload may not be larger than 32767 bytes" in constants:
+            if any([const for const in constants if _is_channel_identifier(const)]):
+                serverbound_packet = class_name
+
+    assert (ignore_clientbound or clientbound_packet is not None) and (ignore_serverbound or serverbound_packet is not None),\
+        f"Unable to find required custom payload packets (client: {clientbound_packet}, server: {serverbound_packet})"
+    
+    return [clientbound_packet, serverbound_packet]
+
+def _get_nethandlers(aggregate, classloader):
+    client_nethandler = aggregate["classes"]["nethandler.client"]
+    server_nethandler = None
+
+    for class_name in classloader.classes:
+     
+        constants = _get_class_constants(classloader, class_name, lambda c: c == " just tried to change non-editable sign")
+        if len(constants) > 0:
+            server_nethandler = class_name
+            break
+
+    assert client_nethandler is not None and server_nethandler is not None,\
+        f"Unable to find both net handlers (client: {client_nethandler}, server: {server_nethandler})"
+
+    return [client_nethandler, server_nethandler]
+
+def _get_class_constants(classloader, class_name, filter_function = lambda c: True):
+    constants = [constant.string.value for constant in classloader.search_constant_pool(path=class_name, type_=String)]
+    return list(filter(filter_function, constants))
```

Comment:
This seems like something that should be in identify.py instead. I assume `nethandler.server` doesn't match this purpose? (`nethandler.server` is probably currently the handshake one, at least for versions after the netty rewrite where there is a dedicated handshake one; it might make sense to rename it to `nethandler.handshake` and use `nethandler.server` for this).

Reply:
Yes, `nethandler.server` is the handshake one, not only for post netty, but every version. It should be simple to identify the play nethandler, since the sign edit message is consistent for all versions (as far as I checked).

Should it be `nethandler.handshake` and `nethandler.server`, or something more specific like `nethandler.server.handshake` and `nethandler.server.play`?

Reply:
I'm pretty sure there isn't a client handshake nethandler since anything the client sends will immediately change from handshake to a different state (and thus the client can't receive anything in the handshaking state). So if that's the case, just `nethandler.handshake` and `nethandler.server` should be fine.